### ISSUE TITLE
Add support for rounded borders

### ISSUE THREAD

**Issue by skynavga** [w3c/ttml2#176]
The use of rounded borders (or backgrounds) on caption/subtitle regions is becoming prominent. With the addition of ``tts:border`` in TTML2, we should also add support for rounded borders basing this support on features defined by CSS Backgrounds and Border Level 3 [1]. Note that this feature may also be used to indicate the use of rounding with background coloration applied to regions or content elements, even when no border is drawn (since the default clip region for the _background painting area_ is the border region).

[1] https://drafts.csswg.org/css-backgrounds-3/#the-border-radius

The proposal is to add an optional ``<border-radii>`` component to the value syntax for ``tts:border`` as follows:

<pre>
tts:border
  : [ &lt;border-thickness&gt; || &lt;border-style&gt; || &lt;border-color&gt; || &lt;border-radii&gt; ]

&lt;border-radii&gt;
  : 'radii(' &lt;length&gt; ( ',' &lt;length&gt; )? ')'
</pre>

where, if two lengths are specified, then the first length corresponds to the quarter ellipse radius for the inline progression dimension (for content) or horizontal direction (for regions), while the second length, if present, corresponds to the quarter ellipse radius for the block progression dimension (for content) or vertical direction (for regions), or, if only one length is present, then it is interpreted as if two lengths were specified with the same value.

If a percentage value is used for a ``<length>``, then it is resolved with respect to the appropriate dimension of the border box's extent.

**Comment by nigelmegitt** [comment]
None of the examples of rounded borders I have seen use different radii for the two directions, so I'd advocate simplifying to a single value for now, until a need for two radii is presented.

However one thing that may be underspecified at the moment is the flow of rounded rectangles between adjacent areas. I've seen this kind of thing:
![image](https://cloud.githubusercontent.com/assets/4988915/18413555/0d647110-77a3-11e6-8f7d-49f1f920b0b0.png)


where I would rather have seen something like this:
![image](https://cloud.githubusercontent.com/assets/4988915/18413553/05a8c354-77a3-11e6-9815-af3746928234.png)

or even perhaps:

![image](https://cloud.githubusercontent.com/assets/4988915/18413552/f194230e-77a2-11e6-9a97-1c660f0d0a3b.png)
 
In other words when creating a composite background shape from congruent area backgrounds the radial centre should be offset in the direction of the hypotenuse of the right angle triangle formed between the vertex being drawn and the two adjacent vertices, not away from it. There are probably more elegant ways to describe this!

**Comment by skynavga** [comment]
I believe it is better to specify specify two separate radii, whether or not implementation support it from the start. I will define a fallback presentation that maps two radii to one radius in the case an implementation doesn't support two.

**Comment by nigelmegitt** [comment]
Further examples from f2f meeting:
![image](https://cloud.githubusercontent.com/assets/4988915/18634866/c8d12f64-7e7a-11e6-97c9-e86b9b63ab99.png)

and 

![image](https://cloud.githubusercontent.com/assets/4988915/18634873/d04b0d00-7e7a-11e6-84c8-860bb1dcb600.png)

These require a little more thought in terms of how to specify them

**Comment by skynavga** [comment]
Committed.
